networkd/network_monitor_thread.ml

```diff
@@ -299,13 +299,28 @@ let ip_watcher () =
     end;
     loop ()
   in
-  while true do
-    try
-      info "(Re)started IP watcher thread";
-      loop ()
-    with e ->
-      warn "Error in IP watcher: %s\n%s" (Printexc.to_string e) (Printexc.get_backtrace ())
-  done
+  try
```

Comment:
I think that the idea is good in general, but the function is getting a little confusing now, with the infinite loop as well as the recursive call, and the double exception handler. I think we just need to remove the while loop and collapse the exception handlers.

Reply:
The while loop seems not feasible to remove. I now removed one level of exception handler.